brainiak/eventseg/event.py

```diff
@@ -489,3 +538,143 @@ def model_prior(self, t):
+        function attempts to find
+        a) pairs of neighboring events that are highly similar, to merge
+        b) events that can be split into two dissimilar events
+        It then tests to see whether simultaneously merging one of the
+        pairs from (a) and splitting one of the events from (b) can improve
+        the log-likelihood. The number of (a)/(b) pairs tested is determined
+        by the split_merge_proposals class attribute.
+
+        Parameters
+        ----------
+        X: list of voxel by time ndarrays
+            fMRI datasets being fit
+
+        log_gamma : list of time by event ndarrays
+            Log probability of each timepoint belonging to each event,
+            for each dataset
+
+        iteration_var : float
+            Current variance in simulated annealing
+
+        curr_ll: float
+            Log-likelihood of current model
+
+        Returns
+        -------
+        return_ll : ndarray with length equal to length of X
+            Log-likelihood after merge/split (same as curr_ll if no
+            merge/split improved curr_ll)
+
+        return_lg : list of time by event ndarrays
+            Log probability of each timepoint belonging to each event,
+            for each dataset (same as log_gamma if no merge/split
+            improved curr_ll)
+
+        return_mp : voxel by event ndarray
+            Mean patterns of events (after possible merge/split)
+        """
+
+        # Compute current probabilities and mean patterns
+        n_train = len(X)
+        n_dim = X[0].shape[0]
+
+        seg_prob = [np.exp(lg) / np.sum(np.exp(lg), axis=0)
+                    for lg in log_gamma]
+        mean_pat = np.empty((n_train, n_dim, self.n_events))
+        for i in range(n_train):
+            mean_pat[i, :, :] = X[i].dot(seg_prob[i])
+        mean_pat = np.mean(mean_pat, axis=0)
+
+        # Compute correlations between every pair of neighboring
+        # events and their merged pattern. High values of
+        # merge_corr indicate events that are good candidates for
+        # being merged.
+        # Also, compute correlations between every event and each half of
+        # the event, with the midpoint defined as the point where
+        # the cumulative probability (across time) of being in the
+        # event = 0.5. Low values of split_corr indicate events
+        # that are good candidates for being split.
+        merge_corr = np.zeros(self.n_events)
+        merge_pat = np.empty((n_train, n_dim, self.n_events))
+        split_corr = np.zeros(self.n_events)
+        split_pat = np.empty((n_train, n_dim, 2 * self.n_events))
+        for i, sp in enumerate(seg_prob):  # Iterate over datasets
+            merge = np.zeros((sp.shape[0], sp.shape[1]))
+            split = np.zeros((sp.shape[0], 2 * sp.shape[1]))
+            cs = np.cumsum(sp, axis=0)
+            for e in range(sp.shape[1]):
+                mid = np.where(cs[:, e] >= 0.5)[0][0]
```

Comment:
This section was the most difficult one for me to follow, partly because of the use of variables named merge and split within the for loop. Merge_corr and merge_pat, for instance, are quite easy to grasp so if there is something you could add to make the purpose of this variable more obvious, I think it would help.

Reply:
I moved and added comments to put them closer to the specific lines of code the referenced and add some more details, and changed the merge/split variables to be more clear.